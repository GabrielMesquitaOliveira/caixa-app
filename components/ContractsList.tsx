import { Contrato, Parcela } from '@/services';
import { formatarMoeda } from '@/utils/formatters';
import React from 'react';
import { Text, View } from 'react-native';

interface ContractsListProps {
  contratos: Contrato[];
  parcelas: Parcela[];
}

export const ContractsList: React.FC<ContractsListProps> = ({ contratos, parcelas }) => {
  return (
    <View className="w-11/12 mx-auto mt-6 mb-8">
      <Text className="text-2xl font-semibold text-gray-800 mb-4">Seus Empr√©stimos</Text>
      
      {contratos.map((contrato) => {
        const parcelasContrato = parcelas.filter(p => p.contratoId === contrato.id);
        const parcelasPagas = parcelasContrato.filter(p => p.situacao === 'paga').length;
        const percentualContrato = parcelasContrato.length > 0 
          ? (parcelasPagas / parcelasContrato.length) * 100 
          : 0;

        return (
          <View key={contrato.id} className="bg-white rounded-xl p-4 shadow-md mb-3">
            <View className="flex flex-row justify-between items-start">
              <View className="flex-1 w-2/3">
                <Text className="text-lg font-semibold text-gray-800">
                  {contrato.nomePersonalizado || contrato.produto?.nome}
                </Text>
                <Text className="text-sm text-gray-600 mt-1">
                  Contrato: {contrato.numeroContrato}
                </Text>
                <Text className="text-sm text-gray-600">
                  Sistema: {contrato.sistemaAmortizacao || 'PRICE'}
                </Text>
              </View>
              <View className="items-end w-1/3">
                <Text className="text-lg font-bold text-blue-600">
                  {formatarMoeda(contrato.valorContratado)}
                </Text>
                <Text className="text-xs text-gray-500">
                  {parcelasPagas}/{contrato.prazoContratado || parcelasContrato.length} parcelas
                </Text>
              </View>
            </View>
            
            <View className="mt-3">
              <View className="w-full bg-gray-200 rounded-full h-2">
                <View 
                  className="bg-blue-600 h-2 rounded-full" 
                  style={{ width: `${percentualContrato}%` }}
                />
              </View>
              <Text className="text-xs text-gray-500 mt-1">
                {Math.round(percentualContrato)}% pago
              </Text>
            </View>
          </View>
        );
      })}
    </View>
  );
};
