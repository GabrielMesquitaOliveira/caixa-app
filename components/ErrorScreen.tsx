import React from 'react';
import { Text, View } from 'react-native';

interface ErrorScreenProps {
  error: string;
}

export const ErrorScreen: React.FC<ErrorScreenProps> = ({ error }) => (
  <View className="h-full w-full flex items-center justify-center bg-neutral-200">
    <Text className="text-lg text-red-600">{error}</Text>
  </View>
);