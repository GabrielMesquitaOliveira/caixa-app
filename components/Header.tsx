import MaterialIcons from '@expo/vector-icons/MaterialIcons';
import type { NativeStackHeaderProps } from '@react-navigation/native-stack';
import React from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

// Estendendo o tipo das options para incluir props customizadas
interface CustomHeaderOptions {
  numeroConta?: string;
  onMenuPress?: () => void;
  onNotificacoesPress?: () => void;
  showAccountNumber?: boolean;
}

interface CustomHeaderProps extends NativeStackHeaderProps {
  options: NativeStackHeaderProps['options'] & CustomHeaderOptions;
}

export function Header({ 
  navigation, 
  route, 
  options,
  back,
}: CustomHeaderProps) {
  // Props padrão ou vindas das options da screen
  const numeroConta = (options as any).numeroConta || "Ag. 3737 CC.576939466-4";
  const showAccountNumber = (options as any).showAccountNumber ?? !back;
  const title = options.title || route.name;

  const handleBackPress = () => {
    if (navigation.canGoBack()) {
      navigation.goBack();
    }
  };

  const handleMenuPress = () => {
    const onMenuPress = (options as any).onMenuPress;
    if (onMenuPress) {
      onMenuPress();
    } else {
      // Comportamento padrão do menu
      console.log('Menu pressed');
    }
  };

  const handleNotificacoesPress = () => {
    const onNotificacoesPress = (options as any).onNotificacoesPress;
    if (onNotificacoesPress) {
      onNotificacoesPress();
    } else {
      // Comportamento padrão das notificações
      console.log('Notificações pressed');
    }
  };

  return (
    <SafeAreaView edges={['top']} className="bg-[#005CA9]">
      <View className="flex-row items-center justify-between px-4 py-3 bg-[#005CA9]">
        {/* Left Side - Menu ou Back Button */}
        <TouchableOpacity 
          onPress={back ? handleBackPress : handleMenuPress} 
          className="p-2"
        >
          <MaterialIcons
            name={back ? "arrow-back" : "menu"}
            size={24}
            color="#fff"
          />
        </TouchableOpacity>

        {/* Center - Numero da conta ou título */}
        <View className="flex-1 items-center justify-center">
          <Text className="text-white text-2xl font-bold">
            {showAccountNumber ? numeroConta : title}
          </Text>
        </View>

        {/* Right Side - Notificações */}
        <TouchableOpacity onPress={handleNotificacoesPress} className="p-2">
          <MaterialIcons
            name="notifications"
            size={24}
            color="#fff"
          />
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}