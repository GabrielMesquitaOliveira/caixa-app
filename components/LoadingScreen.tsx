import React from 'react';
import { ActivityIndicator, Text, View } from 'react-native';

export const LoadingScreen: React.FC = () => (
  <View className="h-full w-full flex items-center justify-center bg-neutral-200">
    <ActivityIndicator size="large" color="#005CA9" />
    <Text className="text-lg text-gray-600 mt-4">Carregando...</Text>
  </View>
);