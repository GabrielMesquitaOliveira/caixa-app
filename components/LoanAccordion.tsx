import {
  Accordion,
  AccordionContent,
  AccordionContentText,
  AccordionHeader,
  AccordionIcon,
  AccordionItem,
  AccordionTitleText,
  AccordionTrigger
} from '@/components/ui/accordion';
import { ChevronDownIcon } from '@/components/ui/icon';
import { formatarMoeda, formatarPorcentagem } from '@/utils/formatters';
import React from 'react';
import { View } from 'react-native';

interface LoanAccordionProps {
  contrato: {
    valorContratado: number;
    dataContratacao: string;
    sistemaAmortizacao: string;
    taxaJurosMensal: number;
    taxaJurosAnual: number;
    prazoContratado: number;
    parcelasPagas: number;
    saldoDevedor: number;
    dataUltimoVencimento: string;
    cetMensal?: number;
    cetAnual?: number;
  };
}

const LoanAccordion: React.FC<LoanAccordionProps> = ({ contrato }) => {
  const prazoRemanescente = contrato.prazoContratado - contrato.parcelasPagas;
  
  return (
    <Accordion variant="filled" size="md" className="mb-3">
      <AccordionItem value="details">
        <AccordionHeader>
          <AccordionTrigger>
            <AccordionTitleText>Detalhes do Empréstimo</AccordionTitleText>
            <AccordionIcon as={ChevronDownIcon} />
          </AccordionTrigger>
        </AccordionHeader>
        <AccordionContent>
          <View className="space-y-2">
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Valor Contratado:</AccordionContentText>
              <AccordionContentText className="font-semibold">{formatarMoeda(contrato.valorContratado)}</AccordionContentText>
            </View>
            
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Data de Contratação:</AccordionContentText>
              <AccordionContentText className="font-semibold">
                {new Date(contrato.dataContratacao).toLocaleDateString('pt-BR')}
              </AccordionContentText>
            </View>
            
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Sistema de Amortização:</AccordionContentText>
              <AccordionContentText className="font-semibold">{contrato.sistemaAmortizacao}</AccordionContentText>
            </View>
            
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Taxa de Juros Mensal:</AccordionContentText>
              <AccordionContentText className="font-semibold">{formatarPorcentagem(contrato.taxaJurosMensal)}</AccordionContentText>
            </View>
            
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Taxa de Juros Anual:</AccordionContentText>
              <AccordionContentText className="font-semibold">{formatarPorcentagem(contrato.taxaJurosAnual)}</AccordionContentText>
            </View>
            
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Prazo Total (Meses):</AccordionContentText>
              <AccordionContentText className="font-semibold">{contrato.prazoContratado}</AccordionContentText>
            </View>
            
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Prazo Remanescente:</AccordionContentText>
              <AccordionContentText className="font-semibold">{prazoRemanescente} meses</AccordionContentText>
            </View>
            
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Saldo Devedor Atualizado:</AccordionContentText>
              <AccordionContentText className="font-semibold">{formatarMoeda(contrato.saldoDevedor)}</AccordionContentText>
            </View>
            
            <View className="flex-row justify-between">
              <AccordionContentText className="text-gray-600">Data Último Vencimento:</AccordionContentText>
                <AccordionContentText className="font-semibold">
                {new Date(contrato.dataUltimoVencimento).toLocaleDateString('pt-BR')}
                </AccordionContentText>
            </View>
            
            {contrato.cetMensal && (
              <View className="flex-row justify-between">
                <AccordionContentText className="text-gray-600">CET Mensal:</AccordionContentText>
                <AccordionContentText className="font-semibold">{formatarPorcentagem(contrato.cetMensal)}</AccordionContentText>
              </View>
            )}
            
            {contrato.cetAnual && (
              <View className="flex-row justify-between">
                <AccordionContentText className="text-gray-600">CET Anual:</AccordionContentText>
                <AccordionContentText className="font-semibold">{formatarPorcentagem(contrato.cetAnual)}</AccordionContentText>
              </View>
            )}
          </View>
        </AccordionContent>
      </AccordionItem>
    </Accordion>
  );
};

export default LoanAccordion;
