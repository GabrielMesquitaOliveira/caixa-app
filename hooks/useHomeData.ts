import { Contrato, contratosService, Parcela, parcelasService, Usuario, usuariosService } from '@/services';
import { useEffect, useState } from 'react';

export interface EstatisticasContratos {
  totalContratos: number;
  saldoTotalDevedor: number;
  totalPago: number;
  totalPendente: number;
  percentualPago: number;
}

export interface ParcelaComContrato extends Parcela {
  contratoNome?: string;
  contratoNumero?: string;
}

export const useHomeData = (userId: number = 1) => {
  const [usuario, setUsuario] = useState<Usuario | null>(null);
  const [contratos, setContratos] = useState<Contrato[]>([]);
  const [parcelas, setParcelas] = useState<Parcela[]>([]);
  const [parcelasVencendoEsteMes, setParcelasVencendoEsteMes] = useState<ParcelaComContrato[]>([]);
  const [estatisticas, setEstatisticas] = useState<EstatisticasContratos | null>(null);
  const [loading, setLoading] = useState<boolean>(true);
  const [error, setError] = useState<string | null>(null);

  const calcularEstatisticas = (contratos: Contrato[], parcelas: Parcela[]): EstatisticasContratos => {
    const parcelasPagas = parcelas.filter(p => p.situacao === 'paga');
    const parcelasPendentes = parcelas.filter(p => p.situacao === 'pendente');

    const totalPago = parcelasPagas.reduce((sum, p) => sum + (p.valorPago || 0), 0);
    const totalPendente = parcelasPendentes.reduce((sum, p) => sum + p.valorParcela, 0);

    const saldoTotalDevedor = contratos.reduce((total, contrato) => {
      const parcelasContrato = parcelas.filter(p => p.contratoId === contrato.id);
      const ultimaParcela = parcelasContrato
        .sort((a, b) => b.numeroParcela - a.numeroParcela)[0];
      return total + (ultimaParcela?.saldoDevedor || 0);
    }, 0);

    const percentualPago = parcelas.length > 0
      ? (parcelasPagas.length / parcelas.length) * 100
      : 0;

    return {
      totalContratos: contratos.length,
      saldoTotalDevedor,
      totalPago,
      totalPendente,
      percentualPago
    };
  };

  const carregarParcelasVencendoEsteMes = async (contratos: Contrato[]) => {
    try {
      const parcelasVencendo = await parcelasService.buscarVencendoEsteMes();
      
      const parcelasComContrato = await Promise.all(
        parcelasVencendo.map(async (parcela) => {
          try {
            const contrato = await contratosService.buscarPorId(parseInt(parcela.contratoId));
            return {
              ...parcela,
              contratoNome: contrato.nomePersonalizado || contrato.produto?.nome,
              contratoNumero: contrato.numeroContrato
            };
          } catch (err) {
            return parcela;
          }
        })
      );

      setParcelasVencendoEsteMes(parcelasComContrato);
    } catch (err) {
      console.error('Erro ao carregar parcelas vencendo este mês:', err);
    }
  };

  useEffect(() => {
    const carregarDados = async () => {
      try {
        setLoading(true);
        setError(null);

        const [usuarioEncontrado, contratosEncontrados] = await Promise.all([
          usuariosService.buscarPorId(userId),
          contratosService.filtrarPorCliente(userId.toString())
        ]);

        setUsuario(usuarioEncontrado);
        setContratos(contratosEncontrados);

        // Buscar parcelas de todos os contratos
        const todasParcelas = await Promise.all(
          contratosEncontrados.map(contrato =>
            parcelasService.buscarPorContrato(contrato.id)
          )
        );

        const parcelasFlat = todasParcelas.flat();
        setParcelas(parcelasFlat);

        // Carregar parcelas vencendo este mês
        await carregarParcelasVencendoEsteMes(contratosEncontrados);

        const stats = calcularEstatisticas(contratosEncontrados, parcelasFlat);
        setEstatisticas(stats);

      } catch (err) {
        console.error('Erro ao carregar dados:', err);
        setError('Erro ao carregar dados da aplicação');
      } finally {
        setLoading(false);
      }
    };

    carregarDados();
  }, [userId]);

  return { 
    usuario, 
    contratos, 
    parcelas, 
    parcelasVencendoEsteMes,
    estatisticas, 
    loading, 
    error 
  };
};
