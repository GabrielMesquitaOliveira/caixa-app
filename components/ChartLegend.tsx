import { EstatisticasContratos } from '@/hooks/useHomeData';
import { formatarMoeda } from '@/utils/formatters';
import React from 'react';
import { Text, View } from 'react-native';

interface ChartLegendProps {
  estatisticas: EstatisticasContratos;
}

export const ChartLegend: React.FC<ChartLegendProps> = ({ estatisticas }) => {
  const dadosLegenda = [
    { name: "Pago", value: estatisticas.totalPago, color: "#22c55e" },
    { name: "Pendente", value: estatisticas.totalPendente, color: "#f59e0b" },
    { name: "Saldo Devedor", value: estatisticas.saldoTotalDevedor, color: "#ef4444" }
  ];

  return (
    <View className="w-11/12 mx-auto mb-8">
      <Text className="text-2xl font-semibold text-gray-800 mb-3">Distribuição Financeira</Text>
      <View className="bg-white rounded-xl p-4 shadow-md">
        {dadosLegenda.map((item, index) => (
          <View key={index} className="flex flex-row items-center justify-between py-2">
            <View className="flex flex-row items-center">
              <View 
                className="w-4 h-4 rounded-full mr-3"
                style={{ backgroundColor: item.color }}
              />
              <Text className="text-sm font-medium text-gray-700">{item.name}</Text>
            </View>
            <Text className="text-sm font-semibold text-gray-800">
              {formatarMoeda(item.value)}
            </Text>
          </View>
        ))}
      </View>
    </View>
  );
};