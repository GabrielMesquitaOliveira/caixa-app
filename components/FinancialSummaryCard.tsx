import { EstatisticasContratos } from '@/hooks/useHomeData';
import { formatarMoeda } from '@/utils/formatters';
import React from 'react';
import { Text, View } from 'react-native';
import { PieChart } from 'react-native-chart-kit';

interface FinancialSummaryCardProps {
  estatisticas: EstatisticasContratos;
}

/**
 * Componente que exibe um resumo financeiro com dados de contratos ativos,
 * saldo devedor total e uma grafico de pizza com a distribui o dos valores
 * pagos, pendentes e saldo devedor.
 * 
 * @example
 * <FinancialSummaryCard estatisticas={estatisticas} />
 * 
 * @param estatisticas - objeto com as estat√≠sticas de contratos ativos
 * @returns JSX.Element
 */
export const FinancialSummaryCard: React.FC<FinancialSummaryCardProps> = ({ estatisticas }) => {
  const dadosGrafico = [
    {
      name: "Pago",
      population: estatisticas.totalPago,
      color: "#22c55e",
      legendFontColor: "#333",
      legendFontSize: 12
    },
    {
      name: "Pendente",
      population: estatisticas.totalPendente,
      color: "#f59e0b",
      legendFontColor: "#333",
      legendFontSize: 12
    },
    {
      name: "Saldo Devedor",
      population: estatisticas.saldoTotalDevedor,
      color: "#ef4444",
      legendFontColor: "#333",
      legendFontSize: 12
    }
  ];

  const chartConfig = {
    backgroundColor: "#ffffff",
    backgroundGradientFrom: "#ffffff",
    backgroundGradientTo: "#ffffff",
    color: (opacity = 1) => `rgba(0, 0, 0, ${opacity})`,
    labelColor: (opacity = 1) => `rgba(0, 0, 0, ${opacity})`,
  };

  return (
    <View className="w-11/12 h-60 mx-auto rounded-2xl bg-white flex flex-row items-center justify-center -mt-20 p-6 shadow-lg">
      <View className="w-2/3 flex flex-col justify-center">
        <Text className="text-xl font-semibold text-gray-700">Saldo Devedor</Text>
        <Text className="text-3xl font-semibold text-red-600">
          {formatarMoeda(estatisticas.saldoTotalDevedor)}
        </Text>
        <Text className="text-sm text-gray-500 mt-1">
          {estatisticas.totalContratos} contrato(s) ativo(s)
        </Text>
      </View>
      <View className="w-1/3 flex flex-col h-full justify-center items-center">
        {dadosGrafico.length > 0 && (
          <PieChart
            data={dadosGrafico}
            width={120}
            height={120}
            chartConfig={chartConfig}
            accessor="population"
            backgroundColor="transparent"
            paddingLeft="0"
            absolute={false}
            hasLegend={false}
          />
        )}
      </View>
    </View>
  );
};
