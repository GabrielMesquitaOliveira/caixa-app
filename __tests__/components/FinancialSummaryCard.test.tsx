import { FinancialSummaryCard } from '@/components/FinancialSummaryCard';
import { EstatisticasContratos } from '@/hooks/useHomeData';
import { render } from '@testing-library/react-native';
import React from 'react';

describe('FinancialSummaryCard Component', () => {
  const mockEstatisticas: EstatisticasContratos = {
    totalContratos: 3,
    saldoTotalDevedor: 15000,
    totalPago: 5000,
    totalPendente: 2000,
    percentualPago: 60
  };

  it('renders correctly with statistics data', () => {
    const { getByText } = render(<FinancialSummaryCard estatisticas={mockEstatisticas} />);
    
    expect(getByText('Saldo Devedor')).toBeTruthy();
    expect(getByText('R$ 15.000,00')).toBeTruthy();
    expect(getByText('3 contrato(s) ativo(s)')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<FinancialSummaryCard estatisticas={mockEstatisticas} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
