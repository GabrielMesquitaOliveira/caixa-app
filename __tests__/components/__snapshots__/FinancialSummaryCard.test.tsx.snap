// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancialSummaryCard Component matches snapshot 1`] = `
<View
  className="w-11/12 h-60 mx-auto rounded-2xl bg-white flex flex-row items-center justify-center -mt-40 p-6 shadow-lg"
>
  <View
    className="w-2/3 flex flex-col justify-center"
  >
    <Text
      className="text-xl font-semibold text-gray-700"
    >
      Saldo Devedor
    </Text>
    <Text
      className="text-3xl font-semibold text-red-600"
    >
      R$Â 15.000,00
    </Text>
    <Text
      className="text-sm text-gray-500 mt-1"
    >
      3
       contrato(s) ativo(s)
    </Text>
  </View>
  <View
    className="w-1/3 flex flex-col h-full justify-center items-center"
  >
    <View
      style={
        {
          "height": 120,
          "padding": 0,
          "width": 120,
        }
      }
    >
      <RNSVGSvgView
        bbHeight={120}
        bbWidth={120}
        focusable={false}
        height={120}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 120,
              "width": 120,
            },
          ]
        }
        width={120}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  [
                    0,
                    -1,
                    1,
                    -1,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="backgroundGradient"
                x1={0}
                x2={120}
                y1={120}
                y2={0}
              />
              <RNSVGLinearGradient
                gradient={
                  [
                    0,
                    436207616,
                    1,
                    436207616,
                  ]
                }
                gradientTransform={null}
                gradientUnits={1}
                name="fillShadowGradientFrom"
                x1={0}
                x2={0}
                y1={0}
                y2={120}
              />
            </RNSVGDefs>
          </RNSVGGroup>
          <RNSVGRect
            fill={
              {
                "payload": 0,
                "type": 0,
              }
            }
            height={120}
            propList={
              [
                "fill",
              ]
            }
            rx={0}
            ry={0}
            width="100%"
            x={0}
            y={0}
          />
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                50,
                60,
              ]
            }
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGPath
                d="M 0 -48 A 48 48 0 0 1 47.511429 -6.831112 L 0 0 A 0 0 0 0 0 0 0 Z "
                fill={
                  {
                    "payload": 4280468830,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGPath
                d="M 47.511429 -6.831112 A 48 48 0 0 1 43.662336 19.939921 L 0 0 A 0 0 0 0 0 0 0 Z "
                fill={
                  {
                    "payload": 4294286859,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGPath
                d="M 43.662336 19.939921 A 48 48 0 1 1 -0.0048 -48 L 0 0 A 0 0 0 1 0 0 0 Z "
                fill={
                  {
                    "payload": 4293870660,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
